### PR TITLE
docs(components): [avatar]

### PR DESCRIPTION
**Pull request by jw-foss**
* Update avatar documentation per new design.
* Move AvatarInstance to instance.ts for better readabilties.

Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: docs/en-US/component/avatar.md [hunk 1]
```diff
@@ -51,21 +51,21 @@ avatar/fit
 
 ### Avatar Attributes
```
Comment (chenxch) [inline docs/en-US/component/avatar.md:52]: Documents with only one component can remove the component name.

File: docs/en-US/component/avatar.md [hunk 2]
```diff
@@ -51,21 +51,21 @@ avatar/fit
...
-| `src`     | the source of the image for an image avatar.              | `string`                                                   | —           | No       |
-| `src-set` | native attribute `srcset` of image avatar.                | `string`                                                   | —           | No       |
-| `alt`     | native attribute `alt` of image avatar.                   | `string`                                                   | —           | No       |
-| `fit`     | set how the image fit its container for an image avatar.  | `'fill' \| 'contain' \| 'cover' \| 'none' \| 'scale-down'` | `'cover'`   | No       |
+| Name      | Description                                               | Type                                                              | Default     | Required |
+| --------- | --------------------------------------------------------- | ----------------------------------------------------------------- | ----------- | -------- |
+| `icon`    | representation type to icon, more info on icon component. | ^[string] / ^[Component]                                          | —           | No       |
```
Comment (chenxch) [inline docs/en-US/component/avatar.md:56]: The highlighting of the name attribute needs to be removed.
`icon` => icon

File: docs/en-US/component/avatar.md [hunk 3]
```diff
@@ -51,21 +51,21 @@ avatar/fit
...
+| `size`    | avatar size.                                              | ^[number] / ^[Enum]`'large' \| 'default' \| 'small'`              | `'default'` | No       |
```
Comment (chenxch) [inline docs/en-US/component/avatar.md:57]: `'default'` => default

**Review by chenxch — COMMENTED** [review]
Consider whether to remove `Required` in this modification.

**Comment by jw-foss** [comment]
@chenxch updated, PTAL.

File: docs/en-US/component/avatar.md [hunk 4]
```diff
@@ -47,28 +47,28 @@ avatar/fit
...
-| `alt`     | native attribute `alt` of image avatar.                   | `string`                                                   | —           | No       |
-| `fit`     | set how the image fit its container for an image avatar.  | `'fill' \| 'contain' \| 'cover' \| 'none' \| 'scale-down'` | `'cover'`   | No       |
+| Name    | Description                                               | Type                                                              | Default     |
+| ------- | --------------------------------------------------------- | ----------------------------------------------------------------- | ----------- |
+| icon    | representation type to icon, more info on icon component. | ^[string] / ^[Component]                                          | —           |
+| size    | avatar size.                                              | ^[number] / ^[Enum]`'large' \| 'default' \| 'small'`              | `'default'` |
+| shape   | avatar shape.                                             | ^[Enum]`'circle' \| 'square'`                                     | `'circle'`  |
```
Comment (chenxch) [inline docs/en-US/component/avatar.md:58]: Enum => enum
Function => function
Everyday types should be kept in lowercase, WDYT?

File: docs/en-US/component/avatar.md [hunk 5]
```diff
@@ -47,28 +47,28 @@ avatar/fit
...
 
-### Avatar Events
+### Events
 
-| Name    | Description                    | Type                 |
-| ------- | ------------------------------ | -------------------- |
-| `error` | trigger when image load error. | `(e: Event) => void` |
+| Name  | Description                    | Type                            |
+| ----- | ------------------------------ | ------------------------------- |
+| error | trigger when image load error. | ^[function]`(e: Event) => void` |
```
Comment (chenxch) [inline docs/en-US/component/avatar.md:68]: function => Function
sry, this is my mistake.

Comment (jw-foss) [reply docs/en-US/component/avatar.md]: @chenxch done, PTAL :bow:

**Review by holazz — APPROVED** [review]

**Review by chenxch — APPROVED** [review]
Well done.

**Pull request merged**